setup.py

```diff
@@ -287,7 +287,14 @@ def run(self):
 
 setup_args.update(setuptools_extra_args)
 
+
+# loose as `.dev` is suppose to be invalid
+loose_pep440re = re.compile('^(\d+)\.(\d+)\.(\d+((a|b|rc)\d+)?)(\.post\d+)?(\.dev\d*)?$')
+
 def main():
+    import IPython.core.release as r
+    if not loose_pep440re.match(r.version):
+        raise ValueError("Version number '%s' is not valid (should match [N!]N(.N)*[{a|b|rc}N][.postN][.devN])" % r.version)
```

Comment:
Do we want to do this check on every use of setup.py, or only when building packages for release?